src/plugins/intel_cpu/src/emitters/x64/jit_snippets_emitters.cpp

```diff
@@ -724,34 +754,27 @@ BrgemmEmitter::BrgemmEmitter(jit_generator* h, cpu_isa_t isa, const ExpressionPt
     std::vector<size_t> leading_dimensions;
     std::vector<std::vector<size_t>> io_layouts;
 
-    auto init_scheduling_params = [&](const std::vector<size_t>& layout, const ov::Shape& io_shape) {
-        if (layout.empty()) {
-            // empty value indicates a planar layout
-            leading_dimensions.push_back(io_shape.back());
-            std::vector<size_t> default_layout(io_shape.size());
-            std::iota(default_layout.begin(), default_layout.end(), 0);
-            io_layouts.push_back(default_layout);
-        } else {
-            // The idea here is to find "2" (for 4D shapes) in the layout and multiply dimensions that are to the right
-            // This implies that "3" is the last layout value, otherwise this layout is not supported.
-            // counting from the end since shape could be prepended with ones
-            const int64_t num_last_dims = layout.end() - std::find(layout.begin(), layout.end(), layout.size() - 2) - 1;
-            if (layout.back() != layout.size() - 1 || num_last_dims < 1)
-                IE_THROW() << "BrgemmEmitter detected invalid layout values: check that this shape + layout combination is schedulable";
-            leading_dimensions.emplace_back(
-                    std::accumulate(io_shape.end() - num_last_dims, io_shape.end(), 1, std::multiplies<size_t>()));
-            io_layouts.push_back(layout);
-        }
+     auto get_layout = [](const std::vector<size_t>& layout, const ov::Shape& io_shape) {
+        if (!layout.empty()) return layout;
+        std::vector<size_t> default_layout(io_shape.size());
+        std::iota(default_layout.begin(), default_layout.end(), 0);
+        return default_layout;
+    };
+
+    auto init_in_scheduling_params = [&](const ov::Input<ov::Node>& input) {
+        io_layouts.push_back(get_layout(snippets::lowered::PortDescriptorUtils::get_port_descriptor_ptr(input)->get_layout(), input.get_shape()));
```

Comment:
Since we pass expressions in emitters, maybe it's worth to pass shape and layout from the expression descriptors as parameters in these methods and avoid OV runtime info usage? I should have done this in scope of my PR, but it seems like I missed this